collections/index.js

```diff
@@ -20,46 +20,126 @@ import employees from './data.json';
   go for it - that's the whole notion of functional programming
 
   The format for this test will be slightly different - the data that I will
-  be using for the test suite lives in ./data.json. I have written you an 
+  be using for the test suite lives in ./data.json. I have written you an
   import statement and you should use these data sets for the test.
 */
+export const howManyEmployees = () => {
+  return employees.length;
+}
 
 /*
   I expect this function to get me an employee by their ID
 */
 export const getEmployeeById = (id) => {
-
+  // created an empty object to insert data into
+  let employee = {};
+  // using a for loop to index through each of the employees on record
+  for (let i = 0; i < employees.length; i++) {
+    // in each of the iterations, if the employee property 'id' matches the
+    // provided arguement 'id', the statement is true...
+    if (employees[i].id === id) {
+      // ... and the following code is executed:
+      // this assigns the employees details matching their id to the
+      // variable 'employee'
+      employee = employees[i];
+    }
+  }
+  // this returns the employee details as the result of the function.
+  return employee;
```

Comment:
There's an array method for this for loop - we're trying to find a single item from an array and return it